### PR TITLE
added narrate command to api

### PR DESCRIPTION
**Pull request by edewit**
@vorburger narrate added

### CODE REVIEW

File: scratch/src/index.ts [hunk 1]
```diff
@@ -63,11 +63,14 @@ let ScratchExtensions: any;
             }
         });
     };
-    ext.narrate = function(entity, text, callack) {
-        eb.send("mcs.actions", { "action": "narrate", "entity": entity, "text": text, "code": code }, function(reply) {
-            callack();
-        });
-
+    ext.narrate = function(entity: string, text: string, callback: Function) {
+        minecraft.narrate(code, entity, text, (err: any) => {
```
Comment (vorburger) [inline scratch/src/index.ts]: I was hoping (see #59) you would have some smart idea to make this re-usable,but let's get back to it later...

File: web/src/main/java/ch/vorburger/minecraft/storeys/api/impl/MinecraftImpl.java [hunk 2]
```diff
@@ -50,15 +53,26 @@ public MinecraftImpl(Vertx vertx, PluginInstance pluginInstance, TokenProvider t
...
+        CompletionStage<Void> completionStage = execute(getPlayer(code), new TitleAction(pluginInstance).setText(Text.of(message)));
         handler.handle(new CompletionStageBasedAsyncResult<>(completionStage));
     }
 
+    @Override
+    public void narrate(String code, String entity, String text, Handler<AsyncResult<Void>> handler) {
+        final NarrateAction narrateAction = new NarrateAction(new Narrator(pluginInstance));
+        narrateAction.setEntity(entity).setText(Text.of(text));
+        execute(getPlayer(code), narrateAction);
+    }
```
Comment (vorburger) [inline web/src/main/java/ch/vorburger/minecraft/storeys/api/impl/MinecraftImpl.java]: I think you missed copy/pasting :rocket: line 57 from above here:

    handler.handle(new CompletionStageBasedAsyncResult<>(completionStage));

File: web/src/test/java/ch/vorburger/minecraft/storeys/web/SeleniumTest.java [hunk 3]
```diff
@@ -51,6 +46,9 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertEquals;
```
Comment (vorburger) [inline web/src/test/java/ch/vorburger/minecraft/storeys/web/SeleniumTest.java]: one fine (future!) day let us agree on what import order we'll want in this project... ;-) and then we can enforce it with Checkstyle (I've done this in a work project) and stop changing this back and forth.

File: web/src/test/java/ch/vorburger/minecraft/storeys/web/test/TestMinecraft.java [hunk 4]
```diff
@@ -34,13 +37,21 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(TestMinecraft.class);
 
-    public String lastTitle;
+    public Map<String, Object> results = new HashMap<>();
```
Comment (vorburger) [inline web/src/test/java/ch/vorburger/minecraft/storeys/web/test/TestMinecraft.java]: Urgh - JS like weakly typed? ;) Can I beg for a `Map<String, String>` at least? :small_airplane: Also how about a `ConcurrentHashMap` here just on the off chance that in the future we ever get crazy enough to want to write tests hammering our shiny new Vert.x RPC concurrently...

File: web/src/test/java/ch/vorburger/minecraft/storeys/web/SeleniumTest.java [hunk 5]
```diff
@@ -137,17 +135,36 @@ public void a_testBasicSetUp() throws Exception {
...
         assertNoBrowserConsoleLogErrors();
-        assertThat(testMinecraft.lastTitle).isEqualTo("hello, world");
+        awaitUntilJSReturnsValue("callback not yet invoked", String.format("return ext.isCallbackCalled('%s')", function));
     }
 
     @SuppressWarnings("unchecked")
     private void awaitUntilJSReturnsValue(String message, String javaScript) {
         // see https://bugs.eclipse.org/bugs/show_bug.cgi?id=534966
-        awaitWD.withMessage(message).until((Function<? super WebDriver, Object>) ExpectedConditions.jsReturnsValue(javaScript));
+        awaitWD.withMessage(message).until(ExpectedConditions.jsReturnsValue(javaScript));
```
Comment (vorburger) [inline web/src/test/java/ch/vorburger/minecraft/storeys/web/SeleniumTest.java]: You (your IDE) probably inadertedly removed this type cast which seemed useless to it, but it breaks the code in Eclipse for me... PITA, I know (see https://bugs.eclipse.org/bugs/show_bug.cgi?id=534966). Is there a way you can mark this method to teach your IDE NOT to touch it automatically?

File: scratch/src/index.ts (hunk 1 above)
Comment (edewit) [reply scratch/src/index.ts]: I was thinking of doing that in an different PR one PR for one change?

Comment (vorburger) [reply scratch/src/index.ts]: sure!

File: web/src/test/java/ch/vorburger/minecraft/storeys/web/SeleniumTest.java (hunk 5 above)
Comment (vorburger) [reply web/src/test/java/ch/vorburger/minecraft/storeys/web/SeleniumTest.java]: FYI I *JUST* sorted this, see 2e729bf4c827eca294383e6b2705d700d8320e64

File: web/src/main/java/ch/vorburger/minecraft/storeys/api/impl/MinecraftImpl.java (hunk 2 above)
Comment (vorburger) [reply web/src/main/java/ch/vorburger/minecraft/storeys/api/impl/MinecraftImpl.java]: was done

File: web/src/test/java/ch/vorburger/minecraft/storeys/web/test/TestMinecraft.java (hunk 4 above)
Comment (vorburger) [reply web/src/test/java/ch/vorburger/minecraft/storeys/web/test/TestMinecraft.java]: was done

**Pull request merged**